lib/chef/provider/package/rubygems.rb

```diff
@@ -482,7 +482,7 @@ def cleanup_after_converge
 
         def candidate_version
           @candidate_version ||= begin
-                                   if target_version_already_installed?(@current_resource.version, @new_resource.version)
+                                   if action != :upgrade && target_version_already_installed?(@current_resource.version, @new_resource.version)
```

Comment:
this seems incorrect, the candidate_version should be the condidate_version no matter :install or :upgrade.

most likely this conditional should just be dropped.

if that breaks :install then there's probably a bug in the superclass to fix (and i think there is at least one around this kind of thing).